modules/core/PrimDisc.h

```diff
@@ -26,6 +26,7 @@ namespace rt {
 			, m_normal(normal)
 			, m_origin(origin)
 			, m_radius(radius)
+               , m_t(Mat::eye(4, 4, CV_32FC1))
 		{}
```

Comment:
Another error, you did not initialize your pivot point. You need to add the line  
`for (int i = 0; i < 3; i++) m_t.at<float>(i, 3) = m_origin[i];` 
in constructor